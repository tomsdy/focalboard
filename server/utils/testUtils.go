package utils

import "github.com/stretchr/testify/mock"

var Anything = mock.MatchedBy(func(interface{}) bool { return true })
