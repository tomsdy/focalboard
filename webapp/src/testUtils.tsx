// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IntlProvider} from 'react-intl'
import React from 'react'

export const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

export function mockDOM(): void {
    window.focus = jest.fn()
    document.createRange = () => {
        const range = new Range()

        range.getBoundingClientRect = jest.fn()

        range.getClientRects = () => {
            return {
                item: () => null,
                length: 0,
                [Symbol.iterator]: jest.fn(),
            }
        }

        return range
    }
}
