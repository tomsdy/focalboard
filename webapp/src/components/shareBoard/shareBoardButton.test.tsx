// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {render} from '@testing-library/react'
import React from 'react'

import {TestBlockFactory} from '../../test/testBlockFactory'
import {wrapDNDIntl} from '../../testUtils'

import ShareBoardButton from './shareBoardButton'

jest.useFakeTimers()

const boardId = '1'

const board = TestBlockFactory.createBoard()
board.id = boardId

describe('src/components/shareBoard/shareBoard', () => {
    test('should match snapshot', async () => {
        const result = render(
            wrapDNDIntl(
                <ShareBoardButton
                    boardId={board.id}
                    enableSharedBoards={true}
                />))

        const renderer = result.container

        expect(renderer).toMatchSnapshot()
    })
})
