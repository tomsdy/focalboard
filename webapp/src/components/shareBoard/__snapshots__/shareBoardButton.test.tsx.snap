// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/shareBoard/shareBoard should match snapshot 1`] = `
<div>
  <div
    class="ShareBoardButton"
  >
    <button
      class="Button emphasis--primary size--medium"
      title="Share board"
      type="button"
    >
      <i
        class="CompassIcon icon-globe CompassIcon"
      />
      <span>
        Share
      </span>
    </button>
  </div>
</div>
`;
